test/rules-google-enhanced-ecommerce-ga4-fullstory.spec.ts

```diff
@@ -0,0 +1,616 @@
+import 'mocha';
+
+import { expect } from 'chai';
+import { expectEqual } from './utils/mocha';
+import {
+  RulesetTestHarness,
+  getRulesetTestEnvironments,
+} from './utils/ruleset-test-harness';
+
+import '../rulesets/google-enhanced-ecommerce-ga4.js';
+
+const ecommerceRulesKey = '_dlo_rules_google_ec_ga4';
+const ecommerceRules = (window as Record<string, any>)[ecommerceRulesKey];
+declare global {
+  // eslint-disable-next-line no-var, vars-on-top
+  var dataLayer: any[];
+}
+
+describe('Ruleset: Google Analytics Enhanced Ecommerce (GA4) to FullStory', () => {
+  getRulesetTestEnvironments().forEach((testEnv) => {
+    describe(`test environment: ${testEnv.name}`, () => {
+      let testHarness: RulesetTestHarness;
+
+      beforeEach(async () => {
+        testHarness = await testEnv.createTestHarness(ecommerceRules, {
+          dataLayer: [],
+        });
+      });
+
+      afterEach(async () => {
+        await testHarness.tearDown();
+      });
+
+      after(async () => {
+        await testEnv.tearDown();
+      });
+
+      /*
+      Event reference:
+      https://developers.google.com/analytics/devguides/collection/ga4/reference/events?client_type=gtm#select_item
+
+      Examples:
+      https://developers.google.com/analytics/devguides/collection/ga4/ecommerce?client_type=gtm#select_an_item_from_a_list
+      */
+      it('reads select_item gtm event', async () => {
+        await testHarness.execute(() => {
+          globalThis.dataLayer.push({
+            event: 'select_item',
+            ecommerce: {
+              items: [
+                {
+                  item_id: 'sku_123',
+                  item_name: 'first item',
+                  price: '1.23',
+                },
+                {
+                  item_id: 'sku_456',
+                  item_name: 'second item',
+                  price: '4.56',
+                },
+              ],
+              item_list_name: 'some list',
+            },
+          });
+        });
+
+        const [eventName, eventProps] = await testHarness.popEvent();
+        expectEqual(eventName, 'select_item');
+        expectEqual(eventProps.item_id, 'sku_123');
+        expectEqual(eventProps.item_name, 'first item');
+        expectEqual(eventProps.price, 1.23);
+        expect(eventProps.item_list_name).to.be.undefined;
```

Comment:
Would we want to capture root level event properties like `item_list_name`? This can also be defined on individual `items`.